packages/aws-helpers/src/stagedRollouts/utils/loadConfig.ts

```diff
@@ -0,0 +1,72 @@
+import { get } from "https";
+import { CloudFrontRequestEvent } from "~/lambdaEdge";
+import { configPath } from "./common";
+import { logDebug } from "./log";
+
+// Config is locally cached within live lambda for a short time (1 minute).
+// Config must be cached per domain.
+// Otherwise cache will spill over different apps, because we may share this lambda.
+const configCache = new Map<string, GatewayConfigCache>();
+
+interface GatewayConfigCache {
+    config: GatewayConfig;
+    timestamp: number;
+}
+
+export interface VariantConfig {
+    domain: string;
+    weight: number;
+}
+
+export type GatewayConfig = Record<string, VariantConfig>;
+
+export async function loadConfig(event: CloudFrontRequestEvent) {
+    const domain = event.Records[0].cf.config.distributionDomainName;
+
+    let config = configCache.get(domain);
+    if (!config || isCacheExpired(config.timestamp)) {
+        logDebug("No config in cache");
+        config = {
+            config: await loadConfigCore(domain),
+            timestamp: Date.now()
+        };
+
+        configCache.set(domain, config);
+    }
+
+    return config.config;
+}
+
+function loadConfigCore(domain: string) {
+    return new Promise<GatewayConfig>((resolve, reject) => {
+        let dataString = "";
+
+        const req = get(
```

Comment:
No, it will not be replaced by WCP endpoint. It's a little bit more complicated - I explained config caching here: https://www.notion.so/webiny/How-traffic-config-is-cached-2c8db57ca2b547a2b2fb1adf378cd191